yapapi/executor/agreements_pool.py

```diff
@@ -0,0 +1,118 @@
+import asyncio
+from dataclasses import dataclass
+import datetime
+import random
+from typing import Dict, NamedTuple, Optional
+from yapapi.executor import events
+from yapapi.props import NodeInfo
+from yapapi.rest.market import Agreement, OfferProposal
+
+
+class _BufferedProposal(NamedTuple):
+    ts: datetime.datetime
+    score: float
+    proposal: OfferProposal
+
+
+@dataclass
+class BufferedAgreement:
+    agreement: Agreement
+    worker_task: Optional[asyncio.Task]
+    has_multi_activity: bool
```

Comment:
I don't like the hackish nature of `dataclasses` package, that's why I decided to leave it as `NamedTuple` as it was in `executor.__init__`. I've used `@dataclass` only because I needed something mutable.